src/GafferScene/EvaluateLightLinks.cpp

```diff
@@ -112,12 +144,16 @@ IECore::ConstCompoundObjectPtr EvaluateLightLinks::computeAttributes( const Scen
 	CompoundObjectPtr result = new CompoundObject;
 	result->members() = inputAttributes->members();
 
-	IECore::StringVectorDataPtr lightNames = new IECore::StringVectorData();
-	std::vector<std::string> &lightNamesWritable = lightNames->writable();
-	PathMatcher linkedlightsSet = SetAlgo::evaluateSetExpression( expressionData->readable(), inPlug() );
-	linkedlightsSet = linkedlightsSet.intersection( inPlug()->set( "__lights" )->readable() );
```

Comment:
We're now potentially computing this twice - let's just compute it once and pass it into `expressionToLightNames()` instead of `in`.

Reply:
I believe I addressed all feedback. Here, `in` is still needed for `evaluateSetExpression`, though? I've kept it for now and just moved the `__lights` set computation out of the function.